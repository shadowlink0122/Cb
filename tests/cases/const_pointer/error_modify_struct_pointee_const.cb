// エラーテスト: const Point* 経由で構造体メンバーを変更しようとする
include "stdio.cb"

struct Point {
    int x;
    int y;
};

int main() {
    println("=== Test: Cannot modify struct through pointer to const ===");
    Point p;
    p.x = 100;
    p.y = 200;
    
    const Point* ptr = &p;
    
    // エラー: const Point* 経由ではメンバーを変更できない
    (*ptr).x = 150;  // This should fail
    
    println("ERROR: This should not execute");
    return 0;
}
