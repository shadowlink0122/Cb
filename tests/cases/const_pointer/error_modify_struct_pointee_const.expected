=== Test: Cannot modify struct through pointer to const ===
Error: Cannot modify value through pointer to const (const T*)
